// ngx-loader-indicator/esm2020/lib/ngx-loader-indicator.directive.mjs
import { Directive, ElementRef, inject, Input, Renderer2 } from '@angular/core';
import { NGX_LOADER_INDICATOR_CONFIG } from './ngx-loader-indicator.config';
import * as i0 from "@angular/core";
export class NgxLoaderIndicatorDirective {
    constructor() {
        this._config = inject(NGX_LOADER_INDICATOR_CONFIG);
        this._el = inject(ElementRef);
        this._renderer = inject(Renderer2);
    }
    set loader(value) {
        if (!value) {
            this.loaderEl ? this._setStyles(this.loaderEl, { display: 'none' }) : null;
            return;
        }
        const { rotate, hostStyles, loaderStyles, imgStyles, img } = this._config;
        this.loaderEl = this._renderer.createElement('div');
        // tslint:disable-next-line:typedef
        const imgEl = this._renderer.createElement('img');
        imgEl.animate([{ transform: 'rotate(360deg)' }, { transform: 'rotate(0deg)' }], rotate);
        this._renderer.appendChild(this._el.nativeElement, this.loaderEl);
        this._renderer.appendChild(this.loaderEl, imgEl);
        this._setStyles(this._el.nativeElement, hostStyles);
        this._setStyles(this.loaderEl, loaderStyles);
        this._setStyles(imgEl, imgStyles);
        if (!img) {
            return;
        }
        this._renderer.setAttribute(imgEl, 'src', img);
    }
    _setStyles(element, styles) {
        if (!styles) {
            return;
        }
        Object.keys(styles).forEach((key) => {
            this._renderer.setStyle(element, key, styles[key]);
        });
    }
}
NgxLoaderIndicatorDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: NgxLoaderIndicatorDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive });
NgxLoaderIndicatorDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.0.4", type: NgxLoaderIndicatorDirective, isStandalone: true, selector: "[loader]", inputs: { loader: "loader" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.0.4", ngImport: i0, type: NgxLoaderIndicatorDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[loader]',
                    standalone: true,
                }]
        }], propDecorators: { loader: [{
                type: Input
            }] } });
