// ngx-loader-indicator/esm2020/lib/ngx-loader-indicator.providers.mjs
import { makeEnvironmentProviders } from '@angular/core';
import { NGX_LOADER_INDICATOR_CONFIG, INITIAL_CONFIG, initialConfig, NEW_CONFIG, } from './ngx-loader-indicator.config';
export function _configFactory(initConfig, configValue) {
    const loaderStyles = configValue
        ? configValue.loaderStyles
        : null;
    const imgStyles = configValue ? configValue.imgStyles : null;
    const rotate = configValue ? configValue.rotate : null;
    return {
        ...initConfig,
        ...configValue,
        loaderStyles: { ...initConfig.loaderStyles, ...loaderStyles },
        imgStyles: { ...initConfig.imgStyles, ...imgStyles },
        rotate: { ...initConfig.rotate, ...rotate },
    };
}
export function provideNgxLoaderIndicator(configValue) {
    return [
        {
            provide: NEW_CONFIG,
            useValue: configValue,
        },
        {
            provide: INITIAL_CONFIG,
            useValue: initialConfig,
        },
        {
            provide: NGX_LOADER_INDICATOR_CONFIG,
            useFactory: _configFactory,
            deps: [INITIAL_CONFIG, NEW_CONFIG],
        },
    ];
}
export function provideEnvironmentNgxLoaderIndicator(configValue) {
    return makeEnvironmentProviders(provideNgxLoaderIndicator(configValue));
}
